src/patch/parse.js

```diff
@@ -7,41 +7,40 @@ export function parsePatch(uniDiff, options = {}) {
     let index = {};
     list.push(index);
 
-    // Ignore any leading junk
+    // Parse diff metadata
     while (i < diffstr.length) {
-      if (/^(Index:|diff -r|@@)/.test(diffstr[i])) {
+      let line = diffstr[i];
+
+      // File header found, end parsing diff metadata
+      if (/^(\-\-\-|\+\+\+|@@)/.test(line)) {
```

Comment:
I left it because it was set before. We could remove it if you want, but it doesn't hurt and would allow to accept patches without the file headers (I was thinking about propose it in other pull-request).